core/src/blocktree.rs

```diff
@@ -386,47 +413,165 @@ impl Blocktree {
         let mut write_batch = batch_processor.batch()?;
 
-        let mut just_inserted_data_indexes = HashMap::new();
+        let mut just_inserted_data_shreds = HashMap::new();
+        let mut just_inserted_coding_shreds = HashMap::new();
+        let mut erasure_metas = HashMap::new();
         let mut slot_meta_working_set = HashMap::new();
         let mut index_working_set = HashMap::new();
 
-        shreds.iter().for_each(|shred| {
+        shreds.into_iter().for_each(|shred| {
             let slot = shred.slot();
+            let shred_index = u64::from(shred.index());
 
-            let _ = index_working_set.entry(slot).or_insert_with(|| {
+            let index = index_working_set.entry(slot).or_insert_with(|| {
                 self.index_cf
                     .get(slot)
                     .unwrap()
                     .unwrap_or_else(|| Index::new(slot))
             });
-        });
 
-        // Possibly do erasure recovery here
+            if let Shred::Coding(coding_shred) = &shred {
+                let serialized_shred = bincode::serialize(&shred).unwrap();
+                let inserted =
+                    write_batch.put_bytes::<cf::ShredCode>((slot, shred_index), &serialized_shred);
+
+                if inserted.is_ok() {
+                    // This gives the index of first coding shred in this FEC block
+                    // So, all coding shreds in a given FEC block will have the same set index
+                    let set_index =
+                        shred_index.saturating_sub(u64::from(coding_shred.header.position));
+                    let erasure_config = ErasureConfig::new(
+                        coding_shred.header.num_data_shreds as usize,
+                        coding_shred.header.num_coding_shreds as usize,
+                    );
 
-        let dummy_data = vec![];
+                    let erasure_meta =
+                        erasure_metas.entry((slot, set_index)).or_insert_with(|| {
+                            self.erasure_meta_cf
+                                .get((slot, set_index))
+                                .expect("Expect database get to succeed")
+                                .unwrap_or_else(|| ErasureMeta::new(set_index, &erasure_config))
+                        });
 
-        for shred in shreds {
-            let slot = shred.slot();
-            let index = u64::from(shred.index());
+                    if erasure_config != erasure_meta.config {
+                        // ToDo: This is a potential slashing condition
+                        warn!("Received multiple erasure configs for the same erasure set!!!");
+                        warn!(
+                            "Stored config: {:#?}, new config: {:#?}",
+                            erasure_meta.config, erasure_config
+                        );
+                    }
+
+                    erasure_meta.set_size(PACKET_DATA_SIZE);
+                    index.coding_mut().set_present(shred_index, true);
```

Comment:
can we get a better name for this space of variables?  index already has a meaning in shred...